src/CommunityToolkit.Maui.Core/Essentials/FileSaver/IFileSaver.shared.cs

```diff
@@ -21,4 +21,45 @@ public interface IFileSaver
 	/// <param name="stream"><see cref="Stream"/></param>
 	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
 	Task<string> SaveAsync(string fileName, Stream stream, CancellationToken cancellationToken);
+
+	/// <summary>
+	/// Saves a file to a target folder on the file system.
+	/// This method doesn't throw Exception.
+	/// </summary>
+	/// <param name="initialPath">Initial path</param>
+	/// <param name="fileName">File name with extension</param>
+	/// <param name="stream"><see cref="Stream"/></param>
+	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+	async Task<FileSaverResult> SaveSafeAsync(string initialPath, string fileName, Stream stream, CancellationToken cancellationToken)
+	{
+		try
+		{
+			var path = await SaveAsync(initialPath, fileName, stream, cancellationToken);
```

Comment:
Let's check the cancellation before starting the task
```suggestion
			cancellationToken.ThrowIfCancellationRequested();
			var path = await SaveAsync(initialPath, fileName, stream, cancellationToken);
```